[Bug] bsp/cvitek: too many warnings in spi drivers

### RT-Thread Version

master

### Hardware Type/Architectures

bsp/cvitek

### Develop Toolchain

GCC

### Describe the bug

```shell
CC build/drivers/drv_spi.o
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c: In function 'spi_configure':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c:76:14: warning: unused variable 'ret' [-Wunused-variable]
   76 |     rt_err_t ret = RT_EOK;
      |              ^~~
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c: In function 'dw_spi_transfer_one':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c:143:14: warning: unused variable 'txlevel' [-Wunused-variable]
  143 |     uint16_t txlevel = 0;
      |              ^~~~~~~
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c:142:13: warning: unused variable 'imask' [-Wunused-variable]
  142 |     uint8_t imask = 0;
      |             ^~~~~
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c: In function 'spi_xfer':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c:201:13: warning: variable 'ret' set but not used [-Wunused-but-set-variable]
  201 |     int32_t ret = 0;
      |             ^~~
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c: In function 'rt_hw_spi_init':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c:338:30: warning: assignment to 'void *' from 'long unsigned int' makes pointer from integer without a cast [-Wint-conversion]
  338 |         _spi_obj[i].dws.regs = (rt_ubase_t)DRV_IOREMAP((void *)_spi_obj[i].dws.regs, 0x1000);
      |                              ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/drv_spi.c:332:20: warning: unused variable 'dws' [-Wunused-variable]
  332 |     struct dw_spi *dws;
      |                    ^~~
```
以及
```bash
CC build/drivers/libraries/spi/dw_spi.o
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'spi_hw_init':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:50:54: warning: left-hand operand of comma expression has no effect [-Wunused-value]
   50 |         SP_DEBUG_LOG("Detected FIFO size: %u bytes\n", dws->fifo_len);
      |                                                      ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:50:54: warning: statement with no effect [-Wunused-value]
   50 |         SP_DEBUG_LOG("Detected FIFO size: %u bytes\n", dws->fifo_len);
      |                     ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'tx_max':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:83:48: warning: left-hand operand of comma expression has no effect [-Wunused-value]
   83 |     SP_DEBUG_LOG("tx left: %#x, tx room: %#x\n", tx_left, tx_room);
      |                                                ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:83:57: warning: left-hand operand of comma expression has no effect [-Wunused-value]
   83 |     SP_DEBUG_LOG("tx left: %#x, tx room: %#x\n", tx_left, tx_room);
      |                                                         ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:83:17: warning: statement with no effect [-Wunused-value]
   83 |     SP_DEBUG_LOG("tx left: %#x, tx room: %#x\n", tx_left, tx_room);
      |                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:92:31: warning: left-hand operand of comma expression has no effect [-Wunused-value]
   92 |     SP_DEBUG_LOG("temp: %#x\n", temp);
      |                               ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:92:17: warning: statement with no effect [-Wunused-value]
   92 |     SP_DEBUG_LOG("temp: %#x\n", temp);
      |                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'dw_writer':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:102:31: warning: left-hand operand of comma expression has no effect -Wunused-value]
  102 |     SP_DEBUG_LOG("max: %#x \n", max);
      |                               ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:102:17: warning: statement with no effect [-Wunused-value]
  102 |     SP_DEBUG_LOG("max: %#x \n", max);
      |                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'rx_max':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:123:46: warning: left-hand operand of comma expression has no effect -Wunused-value]
  123 |     SP_DEBUG_LOG("data_in_fifo:%u temp: %u\n", data_in_fifo, temp);
      |                                              ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:123:60: warning: left-hand operand of comma expression has no effect -Wunused-value]
  123 |     SP_DEBUG_LOG("data_in_fifo:%u temp: %u\n", data_in_fifo, temp);
      |                                                            ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:123:17: warning: statement with no effect [-Wunused-value]
  123 |     SP_DEBUG_LOG("data_in_fifo:%u temp: %u\n", data_in_fifo, temp);
      |                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'dw_spi_check_status':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:138:21: warning: statement with no effect [-Wunused-value]
  138 |         SP_DEBUG_LOG("RX FIFO overflow detected\n");
      |                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:143:21: warning: statement with no effect [-Wunused-value]
  143 |         SP_DEBUG_LOG("RX FIFO underflow detected\n");
      |                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:148:21: warning: statement with no effect [-Wunused-value]
  148 |         SP_DEBUG_LOG("TX FIFO overflow detected\n");
      |                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'dw_reader':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:164:31: warning: left-hand operand of comma expression has no effect -Wunused-value]
  164 |     SP_DEBUG_LOG("max: %#x \n", max);
      |                               ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:164:17: warning: statement with no effect [-Wunused-value]
  164 |     SP_DEBUG_LOG("max: %#x \n", max);
      |                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'dw_spi_set_polarity_and_phase':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:324:49: warning: left-hand operand of comma expression has no effect -Wunused-value]
  324 |     SP_DEBUG_LOG("set phase and polarity: %#x\n", reg);
      |                                                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:324:17: warning: statement with no effect [-Wunused-value]
  324 |     SP_DEBUG_LOG("set phase and polarity: %#x\n", reg);
      |                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'dw_spi_set_clock':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:334:49: warning: left-hand operand of comma expression has no effect -Wunused-value]
  334 |     SP_DEBUG_LOG("clk div value is: %u, hz:%u\n", div, dws->speed_hz);
      |                                                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:334:54: warning: left-hand operand of comma expression has no effect -Wunused-value]
  334 |     SP_DEBUG_LOG("clk div value is: %u, hz:%u\n", div, dws->speed_hz);
      |                                                      ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:334:54: warning: statement with no effect [-Wunused-value]
  334 |     SP_DEBUG_LOG("clk div value is: %u, hz:%u\n", div, dws->speed_hz);
      |                 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'dw_spi_set_data_frame_len':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:349:53: warning: left-hand operand of comma expression has no effect -Wunused-value]
  349 |         SP_DEBUG_LOG("do not support %u bit data!\n", size);
      |                                                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:349:21: warning: statement with no effect [-Wunused-value]
  349 |         SP_DEBUG_LOG("do not support %u bit data!\n", size);
      |                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:354:45: warning: left-hand operand of comma expression has no effect -Wunused-value]
  354 |     SP_DEBUG_LOG("set data frame len: %#x\n", temp);
      |                                             ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:354:17: warning: statement with no effect [-Wunused-value]
  354 |     SP_DEBUG_LOG("set data frame len: %#x\n", temp);
      |                 ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c: In function 'dw_spi_show_regs':
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:360:38: warning: left-hand operand of comma expression has no effect -Wunused-value]
  360 |     SP_DEBUG_LOG("CTRLR0: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_CTRLR0));
      |                                      ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:361:38: warning: left-hand operand of comma expression has no effect -Wunused-value]
  361 |     SP_DEBUG_LOG("CTRLR1: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_CTRLR1));
      |                                      ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:362:38: warning: left-hand operand of comma expression has no effect -Wunused-value]
  362 |     SP_DEBUG_LOG("SSIENR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_SSIENR));
      |                                      ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:363:35: warning: left-hand operand of comma expression has no effect -Wunused-value]
  363 |     SP_DEBUG_LOG("SER: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_SER));
      |                                   ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:364:37: warning: left-hand operand of comma expression has no effect -Wunused-value]
  364 |     SP_DEBUG_LOG("BAUDR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_BAUDR));
      |                                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:365:38: warning: left-hand operand of comma expression has no effect -Wunused-value]
  365 |     SP_DEBUG_LOG("TXFTLR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_TXFTLR));
      |                                      ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:366:38: warning: left-hand operand of comma expression has no effect -Wunused-value]
  366 |     SP_DEBUG_LOG("RXFTLR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_RXFTLR));
      |                                      ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:367:37: warning: left-hand operand of comma expression has no effect -Wunused-value]
  367 |     SP_DEBUG_LOG("TXFLR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_TXFLR));
      |                                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:368:37: warning: left-hand operand of comma expression has no effect -Wunused-value]
  368 |     SP_DEBUG_LOG("RXFLR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_RXFLR));
      |                                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:369:34: warning: left-hand operand of comma expression has no effect -Wunused-value]
  369 |     SP_DEBUG_LOG("SR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_SR));
      |                                  ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:370:35: warning: left-hand operand of comma expression has no effect -Wunused-value]
  370 |     SP_DEBUG_LOG("IMR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_IMR));
      |                                   ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:371:35: warning: left-hand operand of comma expression has no effect -Wunused-value]
  371 |     SP_DEBUG_LOG("ISR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_ISR));
      |                                   ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:372:37: warning: left-hand operand of comma expression has no effect -Wunused-value]
  372 |     SP_DEBUG_LOG("DMACR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_DMACR));
      |                                     ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:373:39: warning: left-hand operand of comma expression has no effect -Wunused-value]
  373 |     SP_DEBUG_LOG("DMATDLR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_DMATDLR));
      |                                       ^
/home/u/ws/duo/rt-thread/bsp/cvitek/drivers/libraries/spi/dw_spi.c:374:39: warning: left-hand operand of comma expression has no effect -Wunused-value]
  374 |     SP_DEBUG_LOG("DMARDLR: \t0x%08x\n", dw_readl(dws, CVI_DW_SPI_DMARDLR));
```

### Other additional context

_No response_

--- a/bsp/cvitek/drivers/drv_spi.c
+++ b/bsp/cvitek/drivers/drv_spi.c
@@ -73,7 +73,6 @@ static rt_err_t spi_configure(struct rt_spi_device *device, struct rt_spi_config
     RT_ASSERT(device->bus->parent.user_data != RT_NULL);
     RT_ASSERT(cfg != RT_NULL);
 
-    rt_err_t ret = RT_EOK;
     struct _device_spi *spi = (struct _device_spi *)device->bus->parent.user_data;
     struct dw_spi *dws = &spi->dws;
 
@@ -139,9 +138,6 @@ static rt_err_t spi_configure(struct rt_spi_device *device, struct rt_spi_config
 
 static rt_err_t dw_spi_transfer_one(struct dw_spi *dws, const void *tx_buf, void *rx_buf, uint32_t len, enum transfer_type  tran_type)
 {
-    uint8_t imask = 0;
-    uint16_t txlevel = 0;
-
     dws->tx = NULL;
     dws->tx_end = NULL;
     dws->rx = NULL;
@@ -198,7 +194,7 @@ static rt_ssize_t spi_xfer(struct rt_spi_device *device, struct rt_spi_message *
 
     struct _device_spi *spi = (struct _device_spi *)device->bus->parent.user_data;
     struct dw_spi *dws = &spi->dws;
-    int32_t ret = 0;
+    int32_t ret = RT_EOK;
 
     if (message->send_buf && message->recv_buf)
     {
@@ -214,6 +210,14 @@ static rt_ssize_t spi_xfer(struct rt_spi_device *device, struct rt_spi_message *
     {
         ret = dw_spi_transfer_one(dws, RT_NULL, message->recv_buf, message->length, POLL_TRAN);
 
+    } else {
+        return 0;
+    }
+
+    if (ret != RT_EOK)
+    {
+        LOG_E("spi transfer error : %d", ret);
+        return 0;
     }
 
     return message->length;
@@ -329,13 +333,12 @@ static void rt_hw_spi_pinmux_config()
 int rt_hw_spi_init(void)
 {
     rt_err_t ret = RT_EOK;
-    struct dw_spi *dws;
 
     rt_hw_spi_pinmux_config();
 
     for (rt_size_t i = 0; i < sizeof(_spi_obj) / sizeof(struct _device_spi); i++)
     {
-        _spi_obj[i].dws.regs = (rt_ubase_t)DRV_IOREMAP((void *)_spi_obj[i].dws.regs, 0x1000);
+        _spi_obj[i].dws.regs = (void *)DRV_IOREMAP((void *)_spi_obj[i].dws.regs, 0x1000);
 
         _spi_obj[i].spi_bus.parent.user_data = (void *)&_spi_obj[i];
         ret = rt_spi_bus_register(&_spi_obj[i].spi_bus, _spi_obj[i].device_name, &_spi_ops);

--- a/bsp/cvitek/drivers/libraries/spi/dw_spi.c
+++ b/bsp/cvitek/drivers/libraries/spi/dw_spi.c
@@ -14,6 +14,7 @@
  * limitations under the License.
  */
 #include <errno.h>
+#include <rtthread.h>
 #include <string.h>
 #include <stdio.h>
 
@@ -23,9 +24,12 @@
 #include <rthw.h>
 
 #ifdef SPI_DEBUG
-#define SP_DEBUG_LOG    printf
+#define SP_DEBUG_LOG(fmt, ...) \
+    do { \
+        rt_kprintf(fmt, ##__VA_ARGS__); \
+    } while(0)
 #else
-#define SP_DEBUG_LOG
+#define SP_DEBUG_LOG(fmt, ...)
 #endif
 
 /* Restart the controller, disable all interrupts, clean rx fifo */

